packages/protocol-http/src/types.ts

```diff
@@ -0,0 +1,51 @@
+import { AbortSignal } from "@smithy/types";
+
+/**
+ * @public
+ *
+ * A mapping of header names to string values. Multiple values for the same
+ * header should be represented as a single string with values separated by
+ * `, `.
+ *
+ * Keys should be considered case insensitive, even if this is not enforced by a
+ * particular implementation. For example, given the following HeaderBag, where
+ * keys differ only in case:
+ *
+ * ```json
+ *    {
+ *        'x-amz-date': '2000-01-01T00:00:00Z',
```

Comment:
Should we be updating these to pull out amazon-specific documentation like this? (i.e `x-request-date` vs `x-amz-date`?

Reply:
Yeah, no point in leaving that as `amz`.